src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -153,6 +175,56 @@ internal void SnapshotCumulative(int indexSnapshot)
         internal MetricPointsAccessor GetMetricPoints()
             => new(this.metricPoints, this.currentMetricPointBatch, this.batchSize);
 
+        internal void ReclaimMetricPoint(ref MetricPoint metricPoint, int idx)
+        {
+            if (this.free == null)
+            {
+                return; // Once there are no free points left, the free list will be initialized.
+            }
+
+            // Free list is available: add the index of this point to it and remove the tag lookup entry.
+            metricPoint.MarkAsFree();
+
+            this.free.Enqueue(idx);
+
+            var tagList = new KeyValuePair<string, object>[metricPoint.Tags.Count]; // TODO: There must be a better way to do this.
+            var i = 0;
+
+            foreach (var tag in metricPoint.Tags)
+            {
+                tagList[i] = tag;
+            }
```

Comment:
I think you are missing an increment for `i` here.

Reply:
Yes indeed. Nice catch.